lib/msal-node/src/client/ManagedIdentityClient.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    Authority,
+    CacheManager,
+    INetworkModule,
+    Logger,
+} from "@azure/msal-common";
+import { AppService } from "./ManagedIdentitySources/AppService";
+import { AzureArc } from "./ManagedIdentitySources/AzureArc";
+import { CloudShell } from "./ManagedIdentitySources/CloudShell";
+import { Imds } from "./ManagedIdentitySources/Imds";
+import { ServiceFabric } from "./ManagedIdentitySources/ServiceFabric";
+import { CryptoProvider } from "../crypto/CryptoProvider";
+import {
+    ManagedIdentityErrorCodes,
+    createManagedIdentityError,
+} from "../error/ManagedIdentityError";
+import { ManagedIdentityRequest } from "../request/ManagedIdentityRequest";
+import { ManagedIdentityId } from "../config/ManagedIdentityId";
+import { ManagedIdentityResult } from "../response/ManagedIdentityResult";
+
+/*
+ * Class to initialize a managed identity and identify the service.
+ * Original source of code: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs
+ */
+export class ManagedIdentityClient {
+    private static identitySource:
+        | ServiceFabric
+        | AppService
+        | CloudShell
+        | AzureArc
+        | Imds;
+
+    constructor(
+        logger: Logger,
+        cacheManager: CacheManager,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider
+    ) {
+        if (!ManagedIdentityClient.identitySource) {
+            ManagedIdentityClient.identitySource =
+                this.selectManagedIdentitySource(
+                    logger,
+                    cacheManager,
+                    networkClient,
+                    cryptoProvider
+                );
+        }
+    }
+
+    public async sendMSITokenRequest(
```

Comment:
nit: `sendManagedIdentityTokenRequest` for consistency.

Reply:
consistency with what?